lib/request.js

```diff
@@ -40,6 +42,18 @@ module.exports = (options) => {
             });
         });
         request.on('error', reject);
+        request.on('timeout', () => {
+            request.abort();
+            reject(new Error('socket timeout'));
```

Comment:
It would probably be nice if the error contained some context about what happened, otherwise seeing something like this in logs makes for a very hard debugging. Putting whole request info might be a bit tricky as you might have to filter out auth-related stuff, but at least having a host here would help:

```js
reject(new Error(`Socket timeout on request to ${options.host}`));
```